package demo;

message str_list {
    repeated string elements = 1;
}

message num_list {
    repeated uint64 elements = 1;
}

message str_params {
    required uint64   key    = 1;
    required str_list value  = 2;
}

message num_params {
    required uint64   key    = 1;
    required num_list value  = 2;
}

message demo_msg {
    required uint64 src_id      = 1;
    required uint64 cmd         = 2;
    required uint64 seq_num     = 3;
    repeated str_params sparams = 4;
    repeated num_params nparams = 5;
}
